### ISSUE TITLE
Unable to check masterkey

### ISSUE THREAD

**Issue by SpectralXSC** [GuntharDeNiro/BTCT#470]
After updating Gunbot I've suddenly got this issue. Worked fine the first couple hours but then it did this.

https://gyazo.com/b9c194c11cd0aa2df6325364188ae157

Did not change the API or anything. It worked but then suddenly no.

Any suggestions?

**Comment by cryptowally** [comment]
I get this on Kraken approximately half the time. If I stop the bot, wait for at least 60 seconds, then start again, I have a good chance of not getting the error.

**Comment by stabenfeldt** [comment]
I get the same error using **Bittrex**. I just extracted 8.0.3 to a new directory. 
Then I copied the bittrex exchange section from my working 7.0.2 config.
```
  "exchanges": {
    "bittrex": {
      "masterkey": "KEY",
      "mastersecret": "SECRET",
      "key": "KEY",
      "secret": "SECRET"
    },
```
I get this error when I start:

```
➜  8.0.3 git:(master) ✗ ./gunthy-macos
Loading config...
Loading exchange - bittrex
Unable to check master key.
Loading config...
```

Any idea what might be wrong?

**Comment by melissaportales** [comment]
Same error here for my Binance key, it was working after upgrading to 8.0.3 then after a reboot I get the error.
Loading config...
Loading exchange - binance
Unable to check master key.
Loading config...

**Comment by NarKotix404** [comment]
UNABLE TO CHECK MASTER KEY

- Check API and Secret are correct
- Ensure ports 3000, 5000-6000 are open on your firewall
- Check any other connectivity limiting software/hardware (Virus scanners, Modems, Routers)
- Check trading permissions on your API
- Check Pair naming is correct

(Just a few things to try)

**Comment by SpectralXSC** [comment]
I've tried all that and checked the whole WIKI. 

No solution so far.

**Comment by daede86** [comment]
same here.. i've tried a fresh install using gb-gui and it worked.. then back without guy and i get the error..

**Comment by stefanbemelmans** [comment]
I've encountered this as well, on windows, try shutting off gunbot, go to date/time settings and shut off "change time automatically", adjust time to 5 minutes ahead,  turn back on "change time automatically", restart gunbot.

**Comment by elb34st** [comment]
Same error here too fro Kraken, can't pickup the Money flow from the exchange and after a while just goes in a loop trying to validate the master key.... when it did work like a hour prior to....
Tried fresh install only with kraken and one pair.... same issue, tried to change the Bot Delay to like 20, same thing too

**Comment by tt5oh** [comment]
Same thing here, I am on Mac. The GUI said it was running but there was no data, and nothing for hours, just progressively chewing through memory to the point my computer is useless (its a i7 16gb 3 GHz). I checked settings, and there should have been buys/sells..nothing. I opened gunthy-macos and kept getting bot crash. Noticed the path for config was a different location. Moved the config to admin folder then got the message invalid key. Opened config and there was no key, put my key in and saved, now get unable to check master key. Tried the time thing, rebooted Mac about 35 times, fresh installs about same amount. Spent all day (830am EST) with no luck. Is anyone finding out why? Seems pretty useless too me.

**Comment by crypt0troll** [comment]
I am getting this for Binance after migrating Gunbot from Mac to my Windows PC. All other exchanges are working (Bittrex, Polo). The same keys work on the Mac but not on the Windows machine.

The Gunthy terminal window keeps popping up also while the unable to chek masterkey message comes up.

**Comment by mrtrgn1** [comment]
Turn **off** "Set time automatically" in Settings - Time&Language - Date&Time . Change your time to a couple of minutes ahead. Turn **on** "Set time automatically" and wait for the clock to be synced, which should be immediately. Run Gunbot & make some money.

**Comment by cryptowally** [comment]
To reproduce this on Kraken, just about every time... just "Apply settings" in GUI, or update the config.js file.  It causes the problem.  This is on a Linux VPS with ntp time sync so it's not a time syncing problem.

**Comment by stabenfeldt** [comment]
@NarKotix404 I can't find any references to those ports anywhere. Especially the [troubleshooting guide](https://github.com/GuntharDeNiro/BTCT/wiki/Troubleshooting) should mention them, IMHO.
Are you sure the ports need to be open?

**Comment by BigB117** [comment]
I'm having this same issue on Cryptopia.

Loading config...
Loading exchange - cryptopia
Unable to check master key.
Loading config...
Loading exchange - cryptopia
Unable to check master key.

Still no solution? I've opened up ports 3000, 5000-6000, running on a Windows 2012 Server. Running 8.0.3

**Comment by RobWillis** [comment]
I'm seeing this same issue.  Using Kraken exhange using gunbot 8.0.3.
Runs for a while then fails

**Comment by kryptobum** [comment]
im having the same problem with poloniex and version 8.0.3.

**Comment by safecoms** [comment]
Haven't seen any comment in 9 days on this, so is this issue solved? Anybody can help because I installed yesterday and the same error come on all exchanges, I have 4 of them with zeno and a VPS linux
Does anybody have any comment to add that could help?

**Comment by RobWillis** [comment]
For me - I had to increase the BOT delay to 15 seconds.  Apparently the bot was hitting the exchange too frequently and the exchange started refusing requests.

**Comment by safecoms** [comment]
Hi @RobWillis 
you mean the settings in the config.js file on BOT_DELAY = 15?

my understanding is that this value is a multiplier.

**Comment by RobWillis** [comment]
All I know is I tried several values for the BOT delay (2,5,10).  Until I found that 15 was working for me consistently and I stopped seeing that unable to check master key show up (I usually had to let the bot run for an hour or so before that error showed up).
Note: this against Kraken.  I believe different exchanges may need different values (and the number of pairs you are using I think affects things as well).
